server/paymentProviders/transferwise/index.ts

```diff
@@ -240,27 +239,38 @@ async function createExpensesBatchGroup(
     sourceCurrency: connectedAccount.data.currency || host.currency,
   });
 
-  const transferIds = await Promise.all(
-    expenses.map(async expense => {
-      const { transfer } = await createTransfer(connectedAccount, expense.PayoutMethod, expense, batchGroup.id);
-      return transfer.id;
-    }),
-  );
+  try {
+    const transferIds = await Promise.all(
+      expenses.map(async expense => {
+        const { transfer } = await createTransfer(connectedAccount, expense.PayoutMethod, expense, {
+          batchGroupId: batchGroup.id,
+          token,
+        });
+        return transfer.id;
+      }),
+    );
 
-  batchGroup = await transferwise.getBatchGroup(token, profileId, batchGroup.id);
-  const includesEveryTransferCreated =
-    batchGroup.transferIds.every(id => transferIds.includes(id)) && batchGroup.transferIds.length == transferIds.length;
-  if (!includesEveryTransferCreated) {
-    await transferwise.cancelBatchGroup(token, profileId, batchGroup.id, batchGroup.version).catch(console.error);
-    throw new Error('Batch group does not include every transfer created');
-  }
+    batchGroup = await transferwise.getBatchGroup(token, profileId, batchGroup.id);
+    const includesEveryTransferCreated =
+      batchGroup.transferIds.every(id => transferIds.includes(id)) &&
+      batchGroup.transferIds.length === transferIds.length;
+    if (!includesEveryTransferCreated) {
+      throw new Error('Batch group does not include every transfer created');
+    }
 
-  batchGroup = await transferwise.completeBatchGroup(token, profileId, batchGroup.id, batchGroup.version);
+    batchGroup = await transferwise.completeBatchGroup(token, profileId, batchGroup.id, batchGroup.version);
 
-  for (const expense of expenses) {
-    await expense.update({
-      data: { ...expense.data, batchGroup },
-    });
+    await Promise.all(
+      expenses.map(expense =>
+        expense.update({
+          data: { ...expense.data, batchGroup },
+        }),
+      ),
+    );
+  } catch (e) {
+    logger.error(e);
+    await transferwise.cancelBatchGroup(token, profileId, batchGroup.id, batchGroup.version).catch(console.error);
```

Comment:
@kewitz : I am not entirely sure, but doesn't it make more sense to send any exception to the logger itself rather than `console.error`? 🤔 

```suggestion
    await transferwise.cancelBatchGroup(token, profileId, batchGroup.id, batchGroup.version).catch(logger.error);
```